pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -78,24 +78,26 @@ func (t *CycleNodeRequestTransitioner) transitionPending() (reconcile.Result, er
 	kubeNodes = existingKubeNodes
 
 	if len(kubeNodes) == 0 {
-		return t.transitionToHealing(fmt.Errorf("no existing nodes in aws matched selector"))
+		return t.transitionToHealing(fmt.Errorf("no existing nodes in cloud provider matched selector"))
 	}
 
 	// Describe the node group for the request
-	t.rm.LogEvent(t.cycleNodeRequest, "FetchingNodeGroup", "Fetching node group: %v", t.cycleNodeRequest.Spec.NodeGroupName)
-	nodeGroup, err := t.rm.CloudProvider.GetNodeGroup(t.cycleNodeRequest.Spec.NodeGroupName)
+	t.rm.LogEvent(t.cycleNodeRequest, "FetchingNodeGroup", "Fetching node group: %v", t.cycleNodeRequest.GetNodeGroupNames())
+	nodeGroups, err := t.rm.CloudProvider.GetNodeGroups(t.cycleNodeRequest.GetNodeGroupNames())
 	if err != nil {
 		return t.transitionToHealing(err)
 	}
 
+	// get instances inside cloud provider node groups
+	nodeGroupInstances := nodeGroups.Instances()
 	// Do some sanity checking before we start filtering things
 	// Check the instance count of the node group matches the number of nodes found in Kubernetes
-	if len(kubeNodes) != len(nodeGroup.Instances()) {
+	if len(kubeNodes) != len(nodeGroupInstances) {
 		t.rm.LogEvent(t.cycleNodeRequest, "NodeCountMismatch",
 			"Instances in node group: %v, nodes in kube: %v",
```

Comment:
We need to make this error clearer, as it may be hard to determine what node group is mismatched out of a list of node groups

Reply:
can add more information related with this mismatch especially when cloud provider has more nodes than nodes in cluster (len(nodeGroupInstances) > len(kubeNodes)). If a node is detached from an ASG (len(kubeNodes)< len(nodeGroupInstances)), then there will be no information about the node group for that node. But I can add the node instance ID/provider ID in there.

Reply:
added some extra information about nodes mismatch inside the log.